src/Npgsql/NpgsqlCommand.cs

```diff
@@ -633,6 +633,9 @@ Task Prepare(bool async, CancellationToken cancellationToken = default)
         {
             foreach (var batchCommand in InternalBatchCommands)
             {
+                batchCommand.Parameters.HasOutputParameters = false;
```

Comment:
In that case, you can just call `batchCommand.Parameters.ValidateAndBind`, which does pretty much the same thing (it also does validation, but that probably doesn't matter that much) and we can also remove `foreach` below.

Reply:
IIRC the problem is that we can't do ValidateAndGetLength when explicitly preparing, since the values aren't yet set. But I agree we could refactor this and extract some common code - but this is meant as a simple fix to backport to 6.0.x as well.